tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -51,44 +56,94 @@ private static bool IsOurKindOfPrimitive(Type t)
+                    for (var i = 0; i < fields.Length; i++)
+                    {
+                        var field = fields[i];
+
+                        var propertyName = GetPropertyName(field.Name);
+                        if (string.IsNullOrEmpty(propertyName))
+                        {
+                            Log.Warning("ExtractProperties - couldn't extract property name from: {FieldName}", field.Name);
+                            continue;
+                        }
+
+                        var dynMethod = new DynamicMethod(
+                            bodyType + "_get_" + propertyName,
+                            typeof(object),
+                            new[] { typeof(object) },
+                            typeof(ObjectExtractor).Module,
+                            true);
+                        var ilGen = dynMethod.GetILGenerator();
+                        ilGen.Emit(OpCodes.Ldarg_0);
+                        if (bodyType.IsValueType)
+                        {
+                            ilGen.Emit(OpCodes.Unbox_Any, bodyType);
+                        }
+
+                        ilGen.Emit(OpCodes.Ldfld, field);
+                        if (field.FieldType.IsValueType)
+                        {
+                            ilGen.Emit(OpCodes.Box, field.FieldType);
+                        }
+
+                        ilGen.Emit(OpCodes.Ret);
+                        var func = (Func<object, object>)dynMethod.CreateDelegate(typeof(Func<object, object>));
+
+                        fieldExtractors[i] = new FieldExtractor() { Name = propertyName, Type = field.FieldType, Accessor = func };
+                    }
+
+                    TypeToExtractorMap.Add(bodyType, fieldExtractors);
+                }
+            }
+
+            var dictSize = Math.Min(WafConstants.MaxContainerSize, fieldExtractors.Length);
             var dict = new Dictionary<string, object>(dictSize);
 
-            for (var i = 0; i < fields.Length; i++)
+            for (var i = 0; i < fieldExtractors.Length; i++)
             {
                 if (dict.Count >= WafConstants.MaxContainerSize || depth >= WafConstants.MaxContainerDepth)
                 {
                     return dict;
                 }
 
-                var field = fields[i];
+                var fieldExtractor = fieldExtractors[i];
 
-                var propertyName = GetPropertyName(field.Name);
-                if (string.IsNullOrEmpty(propertyName))
+                if (fieldExtractor != null)
                 {
-                    Log.Warning("ExtractProperties - couldn't extract property name from: {FieldName}", field.Name);
-                    continue;
+                    var value = fieldExtractor.Accessor.Invoke(body);
+                    if (Log.IsEnabled(LogEventLevel.Debug))
+                    {
+                        Log.Debug("ExtractProperties - property: {Name} {Value}", fieldExtractor.Name, value);
```

Comment:
nit: I don't think you currently log the body type anywhere, might be good to include that here?

```suggestion
                        Log.Debug("ExtractProperties - property: {BodyType}.{Name} {Value}", bodyType.FullName, fieldExtractor.Name, value);
```